core/embed/rust/src/ui/model_mercury/flow/confirm_summary.rs

```diff
@@ -111,37 +119,59 @@ impl ConfirmSummary {
             FrameMsg::Button(_) => Some(FlowMsg::Info),
         });
 
-        // Menu
-        let content_menu = Frame::left_aligned(
-            "".into(),
-            VerticalMenu::empty()
-                .item(theme::ICON_CHEVRON_RIGHT, "Fee info".into())
-                .item(theme::ICON_CHEVRON_RIGHT, "Account info".into())
-                .danger(theme::ICON_CANCEL, "Cancel sign".into()),
-        )
-        .with_cancel_button()
-        .with_swipe(SwipeDirection::Right, SwipeSettings::immediate())
-        .map(|msg| match msg {
-            FrameMsg::Content(VerticalMenuChoiceMsg::Selected(i)) => Some(FlowMsg::Choice(i)),
-            FrameMsg::Button(_) => Some(FlowMsg::Cancelled),
-        });
-
         // FeeInfo
+        let mut has_fee_info = false;
         let mut fee = ShowInfoParams::new(TR::confirm_total__title_fee.into()).with_cancel_button();
         for pair in IterBuf::new().try_iterate(fee_items)? {
             let [label, value]: [TString; 2] = util::iter_into_array(pair)?;
             fee = unwrap!(fee.add(label, value));
+            has_fee_info = true;
         }
         let content_fee = fee.into_layout()?;
 
         // AccountInfo
+        let mut has_account_info = false;
         let mut account = ShowInfoParams::new(TR::send__send_from.into()).with_cancel_button();
         for pair in IterBuf::new().try_iterate(account_items)? {
             let [label, value]: [TString; 2] = util::iter_into_array(pair)?;
             account = unwrap!(account.add(label, value));
+            has_account_info = true;
         }
         let content_account = account.into_layout()?;
 
+        // Menu
+        let mut menu = VerticalMenu::empty();
+        let mut menu_items = Vec::<usize, 3>::new();
+        if has_fee_info {
+            menu = menu.item(
+                theme::ICON_CHEVRON_RIGHT,
+                TR::confirm_total__title_fee.into(),
+            );
+            unwrap!(menu_items.push(MENU_ITEM_FEE_INFO));
+        }
+        if has_account_info {
+            menu = menu.item(
+                theme::ICON_CHEVRON_RIGHT,
+                TR::address_details__account_info.into(),
+            );
+            unwrap!(menu_items.push(MENU_ITEM_ACCOUNT_INFO));
+        }
+        menu = menu.danger(
+            theme::ICON_CANCEL,
+            cancel_text.unwrap_or(TR::send__cancel_sign.into()),
+        );
+        unwrap!(menu_items.push(MENU_ITEM_CANCEL));
+        let content_menu = Frame::left_aligned("".into(), menu)
```

Comment:
Nit: you can use `TString::empty()` for the title arg.

Reply:
https://github.com/trezor/trezor-firmware/pull/4095/commits/d2cded804019bbb5fe3d646216db582943bd5826#diff-5cb317ed79ea16b58238d2743af510c9f82a27c2764811d83f532f8d8358b6f6R161